lib/src/layer/polygon_layer/label.dart

```diff
@@ -6,42 +6,54 @@ import 'package:flutter_map/src/layer/polygon_layer/polygon_layer.dart';
 import 'package:latlong2/latlong.dart';
 import 'package:polylabel/polylabel.dart';
 
+class PolygonBounds {
+  final Offset min;
+  final Offset max;
+
+  const PolygonBounds(this.min, this.max);
+}
+
 void Function(Canvas canvas)? buildLabelTextPainter({
+  required math.Point<double> mapSize,
   required Offset placementPoint,
-  required List<Offset> points,
-  required String labelText,
+  required PolygonBounds bounds,
+  required TextPainter textPainter,
   required double rotationRad,
   required bool rotate,
-  required TextStyle labelStyle,
   required double padding,
 }) {
-  final textSpan = TextSpan(text: labelText, style: labelStyle);
-  final textPainter = TextPainter(
-    text: textSpan,
-    textAlign: TextAlign.center,
-    textDirection: TextDirection.ltr,
-  )..layout();
+  final dx = placementPoint.dx;
+  final dy = placementPoint.dy;
+  final width = textPainter.width;
+  final height = textPainter.height;
 
-  final dx = placementPoint.dx - textPainter.width / 2;
-  final dy = placementPoint.dy - textPainter.height / 2;
-
-  double maxDx = 0;
-  var minDx = double.infinity;
-  for (final point in points) {
-    maxDx = math.max(maxDx, point.dx);
-    minDx = math.min(minDx, point.dx);
+  // Cull labels where the polygon is still on the map but the label would not be.
```

Comment:
It doesn't look like culling works correctly when the map is rotated. I simply added a `print` statement below line 40, and it keeps printing even when the label is offscreen.

Reply:
Interesting. It's because the offset isn't relative to the camera but rather relative to the rotated MobileTransformationLayer. I'm not sure we currently have functionality to easily translate into screen coordinates. If it wasn't for the MobileTransformationLayer we could work with latLngToScreenPoint.

I guess since culling is merely a best effort optimization, not culling is fine. I wasn't able to produce visual glitches as is but I also couldn't convince myself that it's not generally possible. So the safest would probably to disbable culling when the map is rotated. What do you think?

Reply:
I couldn't see any visual glitches either, so I think I'm happy to just keep it on. We should add a comment though, just to point out that the issue is known.

I also couldn't see that the culling didn't work correctly on a concave polygon (the one in the example app), so I think that comment can be removed.

Reply:
> I couldn't see any visual glitches either, so I think I'm happy to just keep it on. We should add a comment though, just to point out that the issue is known.

Personally, I'd rather go the safe route. I did try extreme aspect ratios and it still didn't glitch, the effectiveness is of this optimization is greatly reduced, i.e. it accidentally culls. If we want this to work we should figure out the correct math, I'm just a bit too baked right now.

> I also couldn't see that the culling didn't work correctly on a concave polygon (the one in the example app), so I think that comment can be removed.

That comment is different though and visible on the example app. That's about the padding, i.e. and when not to draw the label even when the label placement point is on screen. Specifically, the intent was to not draw labels if the label doesn't fit into the polygon. In the example app you can see the label being drawn onto the border of the concave polygon. The original author was trying to prevent that case and for convex polygons it works that way.

Reply:
I've got it now. We'll keep it disabled then just in case, but I think it should be alright.